### ISSUE TITLE
Alpha keeps running despite Parser Error

### ISSUE THREAD

**Issue by KillerLink** [alpha-asp/Alpha#259]
The following shortened/crafted input that triggers a parser error as shown in picture below, but does not halt execution / attempt to solve the given problem when called with: `java -jar alpha.jar -i parser_error.lp`

```
%file: parser_error.lp
input_value(a).
output_value(a).

cycle_max_max(3).
cycle_max(0).
{ cycle_max(N+1) } :- cycle_max(N), cycle_max_max(M), N<M.
cycle_last(N) :- cycle(N), ~cycle(N+1).

value(Z,0) :- input_value(Z).

cycle(0).
{ cycle(N+1) } :- cycle(N), cycle_max(K), N<K.

:- not value(X,N), cycle_max(N), output_value(X).
```
![image](https://user-images.githubusercontent.com/17423668/87036436-190d5180-c1eb-11ea-8ed3-6dd416564640.png)

Alpha version used was downloaded from: `https://github.com/alpha-asp/Alpha/releases/download/v0.5.0/alpha.jar`

**Comment by lorenzleutgeb** [comment]
I suspect that there's some error recovery going on [here](https://github.com/alpha-asp/Alpha/blob/master/src/main/java/at/ac/tuwien/kr/alpha/grounder/parser/ProgramParser.java#L54-L99) but Alpha does not communicate well enough what it is doing (that it was able to recover from the error). Maybe someone can take the time to understand the recovery mechanism and improve it. That code wasn't touched in 3 years. If you run into trouble I can also pick that issue up, but I'm busy with other things these days...

**Comment by AntoniusW** [comment]
Yes, what @lorenzleutgeb  indicated seems to be the culprit. As far as I remember, what we do there is: first run the parser in a simplified mode that is faster but not guaranteed to parse all inputs successfully; if that fails, run the parser a second time in the slower mode correctly parsing all well-formed inputs. It seems that the `~` for strong negation (which is currently not supported by Alpha) trips the first parser run, but it can recover from there and hence won't trigger a second parsing run.
Since Alpha swallows all errors on the first parsing run, this does not halt execution of the solver. Probably the logic there needs some fixing (and the way how/what error listeners are given to Antlr).
